File: datatoolbox.py
from __future__ import print_function
import os
from datetime import datetime

def logg_info(msg, tipo='INFO', status='OK'):
    timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
    print("[%s]--%s--%s--%s" % (tipo, timestamp, msg, status))

def makedir(dirname, by=__file__):
    if not os.path.exists(dirname):
        os.makedirs(dirname)
        logg_info('%s mkdir %s' % (by, dirname), tipo='INFO', status='OK')

File: datatoolbox.rb
require 'fileutils'

def logg_info(msg, tipo='INFO', status='OK')
    timestamp = Time.now.strftime('%Y-%m-%d %H:%M:%S')
    puts "[#{tipo}]--#{timestamp}--#{msg}--#{status}"
end

def makedir(dirname, by=File.basename(__FILE__))
    if not File.exist?(dirname)
        FileUtils.mkdir(dirname)
        logg_info("#{by} mkdir #{dirname}", tipo='INFO', status='OK')
    end
end
